### PR TITLE
Fix IDENTITY-5010

### PR DESCRIPTION
**Pull request by mefarazath**
Add UI validations for add/update bps profile pages.

### CODE REVIEW

File: components/org.wso2.carbon.identity.workflow.impl.ui/src/main/resources/web/workflow-impl/update-bps-profile.jsp [hunk 1]
```diff
@@ -83,6 +83,25 @@
                 passwordField.value="" ;
             }
         }
+
+        function doValidation(){
```
Comment (omindu) [inline components/org.wso2.carbon.identity.workflow.impl.ui/src/main/resources/web/workflow-impl/update-bps-profile.jsp]: Check formatting

File: components/org.wso2.carbon.identity.workflow.impl.ui/src/main/resources/web/workflow-impl/update-bps-profile.jsp [hunk 2]
```diff
@@ -83,6 +83,25 @@
...
+            // validate input fields
+            var inputs = document.getElementsByTagName('input');
+            for (var i = 0; i < inputs.length; ++i) {
+                // we only need to validate input text fields here
+                if(inputs[i].type == "text") {
```
Comment (omindu) [inline components/org.wso2.carbon.identity.workflow.impl.ui/src/main/resources/web/workflow-impl/update-bps-profile.jsp]: Check formatting

File: components/org.wso2.carbon.identity.workflow.impl.ui/src/main/resources/web/workflow-impl/add-bps-profile.jsp [hunk 3]
```diff
@@ -39,17 +39,36 @@
...
+                var inputValue = inputs[i].value;
+                var displayName = inputs[i].getAttribute("displayName");
+                if (inputValue == null || inputValue.replace(/^\s+|\s+$/g,'') == '') {
+                    CARBON.showWarningDialog(displayName + " cannot be empty.");
+                    return false;
+                }
+            }
+
+            // validate profile name for whitelist regex
+            if(!doValidateInput(document.getElementById('profile-name'), "<fmt:message key="error.input.validation.msg"/>")) {
```
Comment (omindu) [inline components/org.wso2.carbon.identity.workflow.impl.ui/src/main/resources/web/workflow-impl/add-bps-profile.jsp]: Check formatting

**Comment by mefarazath** [comment]
This PR depends on improvements done to identity-validate.js in identity framework. Therefore https://github.com/wso2/carbon-identity-framework/pull/548 needs to be merged before this PR is resolved.

**Pull request merged**